Trading/Exchange/binance/binance_exchange.py

```diff
@@ -37,16 +72,166 @@ def get_name(cls):
+                )
+        else:
+            account_types.append(self.INVERSE_TYPE)  # allows to fetch inverse markets
+            if linear_count:
+                exchange_manager.logger.error(
+                    f"For now, due to the inverse and linear portfolio split on Binance Futures, OctoBot only "
+                    f"supports either linear or inverse trading at a time. Ignoring {linear_count} linear "
+                    f"futures trading pair as {inverse_count} inverse futures trading pairs are enabled."
+                )
+        return account_types
+
+    @classmethod
+    def get_supported_exchange_types(cls) -> list:
+        """
+        :return: The list of supported exchange types
+        """
+        return [
+            trading_enums.ExchangeTypes.SPOT,
+            trading_enums.ExchangeTypes.FUTURE,
+        ]
+
     def get_additional_connector_config(self):
-        return {
+        config = {
             ccxt_constants.CCXT_OPTIONS: {
                 "quoteOrderQty": False,  # disable quote conversion
                 "recvWindow": 60000,    # default is 10000, avoid time related issues
             }
         }
+        if self.exchange_manager.is_future:
+            config[ccxt_constants.CCXT_OPTIONS]['fetchMarkets'] = self._futures_account_types
+        return config
+
+    @classmethod
+    def update_supported_elements(cls, exchange_manager):
+        if exchange_manager.is_future:
+            cls.UNSUPPORTED_ORDERS = cls.FUTURES_UNSUPPORTED_ORDERS
+        else:
+            cls.UNSUPPORTED_ORDERS = cls.SPOT_UNSUPPORTED_ORDERS
+
+    async def get_balance(self, **kwargs: dict):
+        if self.exchange_manager.is_future:
+            balance = []
+            for account_type in self._futures_account_types:
+                balance.append(await self.connector.get_balance(**kwargs, subType=account_type))
+            # todo remove this and use both types when exchange-side multi portfolio is enabled
+            return balance[0]   # only returning linear portfolio
+        return await self.connector.get_balance(**kwargs)
+
+    def get_order_additional_params(self, order) -> dict:
+        params = {}
+        if self.exchange_manager.is_future:
+            params["reduceOnly"] = order.reduce_only
+        return params
+
+    async def create_order(self, order_type: trading_enums.TraderOrderType, symbol: str, quantity: decimal.Decimal,
+                           price: decimal.Decimal = None, stop_price: decimal.Decimal = None,
+                           side: trading_enums.TradeOrderSide = None, current_price: decimal.Decimal = None,
+                           reduce_only: bool = False, params: dict = None) -> typing.Optional[dict]:
+        if self.exchange_manager.is_future:
+            # on futures exchange expects, quantity in contracts: convert quantity into contracts
+            quantity = quantity / self.get_contract_size(symbol)
```

Comment:
I'm not sure but it seems to be the way to get the quantity for a linear symbol but no the one for an inverse right?

Reply:
It's working for both according to my tests:
- linear (contract size on binance is always 1 on linear anyway) BTC/USDT:USDT buy 0.1BTC => contract size = 0.0001, quantity= 0.1 /  0.0001 = 1000 
- inverse BTC/USDT:BTC buy 100 USDT=> contract size = 10, quantity=100 / 10 = 10

Reply:
=> and we do the opposite formula when adapting orders from ccxt format to octobot (from contract to settlement asset)

Reply:
Ok thanks :+1: